### PR TITLE
add `getValueByColumnType` method

### PR DESCRIPTION
**Pull request by sunbufu**
Fixes https://github.com/apache/incubator-shardingsphere/issues/2604.

Changes proposed in this pull request:
- add `getValueByColumnType` method
- add some unit test

### CODE REVIEW

File: sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java [hunk 1]
```diff
@@ -68,12 +68,12 @@ public boolean next() throws SQLException {
     
     @Override
     public Object getValue(final int columnIndex, final Class<?> type) throws SQLException {
-        return decrypt(columnIndex, QueryResultUtil.getValue(resultSet, columnIndex));
+        return decrypt(columnIndex, QueryResultUtil.getValue(resultSet, columnIndex, type));
```
Comment (tuohai666) [inline sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

Comment (sunbufu) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: > StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

This is why i doesn't deleted `QueryResultUtil.getValueByColumnType()`, I have overload a `getValue()` method with `Class<?> type` in `QueryResultUtil`,  `StreamQueryResult` will call the new `getValue` with `Class<?> type`, and `MemoryQueryResult ` will continue to call the old `getValue()`.
image
image

Comment (sunbufu) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: > StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

This is why I doesn't deleted `QueryResultUtil.getValueByColumnType()`, I just have overload a `getValue()` method with `Class<?> type` in `QueryResultUtil`,  `StreamQueryResult` will call the new `getValue()` with `Class<?> type`, and `MemoryQueryResult ` will continue to call the old `getValue()`.
StreamQueryResult call the new getValue
MemoryQueryResult call the old getValue

Comment (tuohai666) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: If so, the problem was not resolved. Two modes get two different value is even worse, isn't?

Comment (sunbufu) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: It's my fault, I overlooked this.
And I have a question. Why do we need to create a QueryRow Iterator and manage it ? Why not do just like [druid's DruidPooledResultSet](https://github.com/alibaba/druid/blob/master/src/main/java/com/alibaba/druid/pool/DruidPooledResultSet.java) ？

Comment (tuohai666) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: It's OK. We have noticed your work and expect more contribution from you. I suggest you become a contributor first, then try to become a committer.

> And I have a question. Why do we need to create a QueryRow Iterator in MemoryQueryResult and manage it ? Why not do just like [druid's DruidPooledResultSet](https://github.com/alibaba/druid/blob/master/src/main/java/com/alibaba/druid/pool/DruidPooledResultSet.java) ？

Sometimes we need to get all data to memory:
1. Merge engine:
https://shardingsphere.apache.org/document/current/en/features/sharding/principle/merge/
2. Execute engine:
https://shardingsphere.apache.org/document/current/en/features/sharding/principle/execute/

Comment (sunbufu) [reply sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java]: I'll read it first, thank you.

**Pull request closed without merging**